### PR TITLE
Enhancement: Flow run results lists

### PR DESCRIPTION
**Pull request by znicholasbrown**
This PR:

- improves mock flow run and task run results artifact description generation (they now more closely resemble action descriptions)
- breaks flow run and task run results into their own lists
- improves some `ArtifactCard` designs
- adds result descriptions to the default slot of `ArtifactCard`s 
- adds an interval to the artifact subscription so results update in real time
- persists a user's list vs grid preference (just using local storage right now but that seemed light enough)
- adds some new demo sections

Screen Shot 2023-03-15 at 9 53 59 PM

Screen Shot 2023-03-15 at 9 54 08 PM

### CODE REVIEW

File: src/localization/locale/en.ts [hunk 1]
```diff
@@ -88,6 +88,8 @@ export const en = {
   info: {
     artifact: 'Artifact',
     noResults: 'No tracked results',
+    flowRunResults: 'Flow run',
+    taskRunResults: 'Task runs',
```
Comment (pleek91) [inline src/localization/locale/en.ts:92]: Nit: As we start adding more of these we need to be careful about naming. If we ever add a second language the names will be important. I'd literally name these `flowRun` and `taskRuns` as these are pretty generic labels and aren't restricted to results.

**Comment by pleek91** [comment]
Couple things I noticed while looking at the deploy preview on my phone. The grid view overlaps and doesn't scroll well. And in the list view the breadcrumbs overflow the card

image
image

**Comment by znicholasbrown** [comment]
@pleek91 which browser are you seeing that in? I've got a pretty nice looking version in chrome desktop at 330px:

Screenshot 2023-03-16 at 11 51 41 AM

**Comment by zhen0** [comment]
Small note here that the deploy preview looks good on my iphone now.  Looks like you're digging in further though? Let me know if I can help keep this one moving.

**Comment by znicholasbrown** [comment]
Thanks @zhen0; the remember your selection feature doesn't seem to be working in the deploy preview but is working in a regular ole build. I've added some fallbacks so that we can feel comfortable releasing just in case that continues to fail downstream. 

@pleek91 I've added a cache-key wrapper

**Review by pleek91 — APPROVED** [review]

**Pull request merged**